app/lib/actions/pulp/repository/refresh.rb

```diff
@@ -39,7 +39,8 @@ def update_or_associate_importer(capsule_id, repository, repository_details)
         def update_or_associate_distributors(capsule_id, repository, repository_details)
           concurrence do
             existing_distributors = repository_details["distributors"]
-            repository.generate_distributors(capsule_id.present?).each do |distributor|
+            capsule = capsule_id ? ::Katello::CapsuleContent.new(SmartProxy.find(capsule_id)) : nil
```

Comment:
It was mainly because some of the other places using the generate_distributors method already have a capsule_content object. Also, in general it also looks like many of the methods in this section of code use 'capsule' they are using a capsule_content object so I was trying to avoid confusion. Though I hear what you're saying. I'll go ahead and change it.